hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutAllOperation.java

```diff
@@ -131,18 +147,14 @@ private Data getValueOrPostProcessedValue(Data dataKey, Data dataValue) {
         return mapServiceContext.toData(record.getValue());
     }
 
-    private boolean shouldWanReplicate() {
-        return mapContainer.getWanReplicationPublisher() != null && mapContainer.getWanMergePolicy() != null;
-    }
-
     @Override
     public Object getResponse() {
         return true;
     }
 
     @Override
     public boolean shouldBackup() {
-        return !backupEntries.isEmpty();
+        return (hasBackups && !backupEntries.isEmpty());
```

Comment:
is this needed? when `hasBackup` is `false` then `backupEntries` is always empty, isn't it?